src/ui/public/courier/fetch/request/search_request/search_request.js

```diff
@@ -27,38 +27,65 @@ export function SearchRequestProvider(Promise) {
       if (!errorHandler) {
         throw new Error('errorHandler is required');
       }
+
       this.errorHandler = errorHandler;
       this.source = source;
       this.defer = defer || Promise.defer();
       this.abortedDefer = Promise.defer();
       this.type = 'search';
+
+      // Track execution time.
+      this.moment = undefined;
+      this.ms = undefined;
+
+      // Lifecycle state.
+      this.started = false;
+      this.stopped = false;
+      this._isFetchRequested = false;
+
       searchRequestQueue.add(this);
     }
 
     /**
-     *  Called by the loopers to find requests that should be sent to the
-     *  fetch() module. When a module is sent to fetch() it's _fetchRequested flag
-     *  is set, and this consults that flag so requests are not send to fetch()
+     *  Called by the searchPoll to find requests that should be sent to the
+     *  fetchSoon module. When a module is sent to fetchSoon its _isFetchRequested flag
+     *  is set, and this consults that flag so requests are not send to fetchSoon
      *  multiple times.
      *
      *  @return {Boolean}
      */
     canStart() {
-      return !this._fetchRequested && !this.stopped && !this.source._fetchDisabled;
+      if (this.source._fetchDisabled) {
+        return false;
+      }
+
+      if (this.stopped) {
```

Comment:
I don't suppose in your travels of courier you understand the logic behind this?  I like how you split this out, but it logically seems strange to me.  No worries if you don't since this PR is unrelated, but if you do know, I think this could go for a comment.  Why can't this be started if it's stopped?  you can't "restart"?  And below, I guess `this._isFetchRequested` means that it's already been started?  The top one is obvious.

Reply:
Funny you should ask! I made these comments as I started trying to make sense of all of this but then I gave up. A lot of this logic exists to support courier's indirection-based architecture and from everything I've seen, I think the best solution will be to replace it with a more explicit architecture (e.g. https://github.com/elastic/kibana/issues/20364).

> Why can't this be started if it's stopped? you can't "restart"?

I don't know the answers to these ones.

> And below, I guess this._isFetchRequested means that it's already been started?

I actually read a comment you left in this file which explained this for me:

```js
    /**
     *  Called by the fetchSoon module when this request has been sent to
     *  be fetched. At that point the request is somewhere between `ready-to-start`
     *  and `started`. The fetch module then waits a short period of time to
     *  allow requests to build up in the request queue, and then immediately
     *  fetches all requests that return true from `isFetchRequestedAndPending()`
     *
     *  @return {undefined}
     */
    _setFetchRequested() {
      this._isFetchRequested = true;
    }
```

Reply:
So I guess it means it's going to start, but hasn't started yet?